src/ime.cpp

```diff
@@ -556,6 +564,30 @@ static void manager_handle_create_input_method(struct wl_client *client, struct
 	active_input_method = ime;
 }
 
+struct wlserver_input_method *create_local_ime()
+{
+	struct wlserver_input_method *ime = new wlserver_input_method();
+	ime->resource = nullptr;
+	ime->manager = global_manager;
+	ime->serial = 1;
+	ime->next_keycode_index = 0;
+	ime->held_keycode = -1;
+	ime->held_modifier_mask = 0;
+	ime->prev_mods = wlr_keyboard_modifiers{0};
+
+	wlr_keyboard_init(&ime->keyboard, &keyboard_impl, "local_ime");
+
+	wlr_keyboard_set_repeat_info(&ime->keyboard, 0, 0);
+
+	wl_resource_set_user_data(ime->resource, ime);
+	gamescope_input_method_send_done(ime->resource, ime->serial);
```

Comment:
I assume this is WIP, but as `ime->resource` is never changed to something other than `nullptr` this segfaults for me

Reply:
Should be fixed now.